design/one-pager-pluggable-secret-stores.md

```diff
@@ -0,0 +1,233 @@
+# Pluggable Secret Stores
+
+* Owner: Hasan Türken (@turkenh)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+In the [v1.7.0] release, we have introduced support for publishing connection
+details secrets to [External Secret Stores] as an alpha feature. With this
+initial iteration, we have focused on the _in-tree_ implementation of secret
+stores by adding [Hashicorp's Vault] support as our first External Secret Store.
+However, this approach introduces [additional dependencies] to Crossplane core
+and would not scale once the community wants to support more secret stores.
+Even further, it won't be possible to use an in-house secret store with
+Crossplane since adding support for such a store to Crossplane core wouldn't
+make sense. For this purpose, with the initial design, we set it as a [goal] to
+support switching to an out-of-tree (plugin) model but intentionally left it
+[out of scope] to avoid the upfront complexity.
+
+With the gained confidence, we can now iterate with a more scalable option by
+supporting pluggable secret stores.
+
+## Goals
+
+The goal of this document is to propose a pluggable secret store mechanism
+that enables the development and consumption of out-of-tree secret store
+plugins.
+
+This new mechanism should:
+- Allow building/using new secret stores for Crossplane without making any
+changes in Crossplane and/or Provider repositories.
+- Provide a smooth development and consumption experience. 
+- Have secure communication with the plugin process.
+
+## Proposal
+
+This document proposes introducing a plugin mechanism for communicating with
+External Secret Stores. We will follow a similar approach as [KMS plugins] used
+by Kubernetes API Server for encryption at rest. A Secret Store plugin will
+be serving a well-defined API via gRPC. For the sake of security, plugins will
+be deployed as sidecars to Crossplane and Provider pods and communication will
+be over Unix domain sockets.
```

Comment:
It may be worth adding a little information about how two containers within a pod can communicate via a Unix domain socket - I had to lookup some documentation to refresh my memory. I'm pretty sure it doesn't require any special privileges, right? The two containers just need to share the volume where the socket exists.

Reply:
Added a new section with the title: `Communication over Unix Domain Sockets`.